test/timeout_test.go

```diff
@@ -129,33 +129,26 @@ func TestPipelineRunTimeout(t *testing.T) {
 	}
 
 	t.Logf("Waiting for TaskRuns from PipelineRun %s in namespace %s to be cancelled", pipelineRun.Name, namespace)
-	errChan2 := make(chan error, len(taskrunList.Items))
-	defer close(errChan2)
-
 	for _, taskrunItem := range taskrunList.Items {
-		go func() {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```

Comment:
and here :)

```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop